app/views/locations/name.html.erb

```diff
@@ -0,0 +1,39 @@
+<% content_for :title, "Add a location to this scheme" %>
+
+<% content_for :before_content do %>
+  <%= govuk_back_link(
+        text: "Back",
+        href: if params[:referrer] == "check_answers"
+                scheme_location_check_answers_path(@scheme, @location)
+              else
+                params[:referrer] == "details" ? scheme_location_path(@scheme, @location) : scheme_location_postcode_path(@scheme, @location)
```

Comment:
Even if the local authority is explicitly entered (this case should be rare), we go back to the postcode page (which would redirect again to the LA page), the designs weren't clear and it seems unneccessarily fiddly to add a referrer to this route. I think this is fine as MVP but could be persuaded otherwise :)